watertap/unit_models/tests/test_biploar_electrodialysis_0D.py

```diff
@@ -0,0 +1,746 @@
+        m.fs.unit.inlet_acidic.flow_mol_phase_comp[0, "Liq", "OH_-"].fix(7.38e-4)
+
+        m.fs.unit.inlet_basic.flow_mol_phase_comp[0, "Liq", "H2O"].fix(2.40e-1)
+        m.fs.unit.inlet_acidic.flow_mol_phase_comp[0, "Liq", "H2O"].fix(2.40e-1)
+
+        m.fs.unit.shadow_factor.fix(1)
+        m.fs.unit.water_trans_number_membrane["bpem"].fix((5.8 + 4.3) / 2)
+        m.fs.unit.water_permeability_membrane["bpem"].fix((2.16e-14 + 1.75e-14) / 2)
+        m.fs.unit.electrodes_resistance.fix(0)
+        m.fs.unit.current_utilization.fix(1)
+        m.fs.unit.channel_height.fix(2.7e-4)
+        m.fs.unit.membrane_areal_resistance.fix((1.89e-4 + 1.77e-4) / 2)
+        m.fs.unit.cell_width.fix(0.1)
+        m.fs.unit.cell_length.fix(0.79)
+        m.fs.unit.membrane_thickness["bpem"].fix(8e-4)
+
+        # Set scaling of critical quantities.
+        m.fs.properties.set_default_scaling(
+            "flow_mol_phase_comp", 1e3, index=("Liq", "Na_+")
+        )
+        m.fs.properties.set_default_scaling(
+            "flow_mol_phase_comp", 1e3, index=("Liq", "Cl_-")
+        )
+        m.fs.properties.set_default_scaling(
+            "flow_mol_phase_comp", 1e3, index=("Liq", "H_+")
+        )
+        m.fs.properties.set_default_scaling(
+            "flow_mol_phase_comp", 1e3, index=("Liq", "OH_-")
+        )
+        m.fs.properties.set_default_scaling(
+            "flow_mol_phase_comp", 1e2, index=("Liq", "H2O")
+        )
+        m.fs.unit.cell_num.fix(1)
+        iscale.set_scaling_factor(m.fs.unit.k_a, 1e-2)
+        iscale.set_scaling_factor(m.fs.unit.k_b, 1e-4)
+        iscale.set_scaling_factor(m.fs.unit.voltage, 1e-1)
+        iscale.set_scaling_factor(m.fs.unit.flux_splitting, 1e3)
+        return m
+
+    @pytest.mark.unit
+    def test_assign(self, bped):
+
+        # Experimental data from Wilhelm et al. (2002) with additional inputs from Mareev et al. (2020)
+        expt_current_density = np.array(
+            [11.7, 15.5, 20.8, 24.4, 30.6, 40.0, 100.6]
+        )  # in mA/cm2
+        expt_membrane_potential = np.array(
+            [0.088, 0.184, 0.4045, 0.690, 0.858, 0.95, 1.3]
+        )  # in volts
+
+        for indx, v in enumerate(expt_membrane_potential):
+            m = bped
+            m.fs.unit.potential_membrane_bpem[0].fix(v)
+
+            iscale.calculate_scaling_factors(m.fs)
+            assert degrees_of_freedom(m) == 0
+            initialization_tester(m)
+            results = solver.solve(m)
+            assert_optimal_termination(results)
+            if indx < 2:
+                rel_tol = 1e0
+            else:
+                rel_tol = 1.5e-1
+
+            current_density_computed = (
+                0.1
+                * m.fs.unit.current[0]
+                / (m.fs.unit.cell_width * m.fs.unit.cell_length)
+            )  # convert to mA/cm2
+            assert value(current_density_computed) == pytest.approx(
+                expt_current_density[indx], rel=rel_tol
+            )
+
+
```

Comment:
I think the purpose of such tests is to ensure the model is numerically stable over time, i.e., it reproduces key solutions over time.  For such a  purpose, you may wanna set a constant and small tol  to assert the equality to a value that is close within acceptable errors by your estimate to the value you are citing from literatures.